addons/account/wizard/accrued_orders.py

```diff
@@ -0,0 +1,120 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from dateutil.relativedelta import relativedelta
+import json
+from odoo import models, fields, api, _, Command
+from odoo.tools import format_date, float_is_zero
+from odoo.exceptions import UserError
+
+
+class AccruedExpenseRevenue(models.TransientModel):
+    _name = 'account.accrued.orders.wizard'
+    _description = 'Accrued Orders Wizard'
+
+    def _default_journal_id(self):
+        return self.env['account.journal'].search(
+            [('type', '=', 'general'), ('company_id', '=', self.env.company.id)], limit=1
+        ),
+
+    company_id = fields.Many2one('res.company', default=lambda self: self.env.company)
+    journal_id = fields.Many2one(
+        comodel_name='account.journal',
+        default=_default_journal_id,
+        domain=[('type', '=', 'general')],
+        readonly=False,
+        required=True,
+        string='Journal',
+    )
+    date = fields.Date(default=fields.Date.today(), required=True)
+    reversal_date = fields.Date(default=fields.Date.today() + relativedelta(days=1), required=True)
+    account_id = fields.Many2one('account.account', string='Accrued Account', required=True)
+    preview_data = fields.Text(compute='_compute_preview_data')
+
+    @api.depends('date', 'journal_id', 'account_id')
+    def _compute_preview_data(self):
+        for record in self:
+            preview_vals = [self.env['account.move']._move_dict_to_preview_vals(self._compute_move_vals(), record.company_id.currency_id)]
+            preview_columns = [
+                {'field': 'account_id', 'label': _('Account')},
+                {'field': 'name', 'label': _('Label')},
+                {'field': 'debit', 'label': _('Debit'), 'class': 'text-right text-nowrap'},
+                {'field': 'credit', 'label': _('Credit'), 'class': 'text-right text-nowrap'},
+            ]
+            record.preview_data = json.dumps({
+                'groups_vals': preview_vals,
+                'options': {
+                    'columns': preview_columns,
+                },
+            })
+
+    def _compute_move_vals(self):
+        move_lines = []
+        orders = []
+        if self.env.context.get('active_model') == 'purchase.order':
+            orders = self.env['purchase.order'].browse(self.env.context.get('active_ids'))
+            is_purchase = True
+        else:
+            orders = self.env['sale.order'].browse(self.env.context.get('active_ids'))
+            is_purchase = False
+        if len(orders.mapped('company_id')) > 1:
+            raise UserError(_('Entries can only be created for one company at a time.'))
+        for order in orders:
+            amt = order.amount_total - sum(order.invoice_ids.mapped('amount_total'))
```

Comment:
`amount_total` can be in other currency.

`amt = order.amount_total - sum(order.invoice_ids.mapped('amount_total'))` is wrong if you have 1 invoice for 2 pruchases.
You should use `(product_uom_qty - qty-invoiced) * price_unit ` (and get the untaxed amount) for each line.